`no-undefined-types` - false positive on `@link` to symbol declared inside `declare global` block

<!--
NOTE 1: It would help if you can first scan the issue tracker before submitting
a new issue: https://github.com/gajus/eslint-plugin-jsdoc/issues

NOTE 2: If you are running ESLint/`eslint-plugin-jsdoc` from within an IDE,
before filing an issue, please be sure to restart the IDE after making any
updates to ensure you are indeed getting errors with the updated version.
Running `eslint` on the command line may also be helpful in allowing you
to better provide us with the error details including the stack trace.

NOTE 3: If you have a bug specifically related to the parsing of types, e.g.,
the `some-unrecognized+type-syntax` (within curly brackets) in:

/**
 * @param {some-unrecognized+type-syntax} myName
 */

...then please file instead at https://github.com/simonseyock/jsdoc-type-pratt-parser/issues/
-->

<!--
TIP: If you have a complicated config with `overrides`, you can simplify the
config you provide us by running `eslint` with `--print-config` and the file
that is triggering the error. See:
https://eslint.org/docs/user-guide/command-line-interface#print-config
-->

## Expected behavior
no error

## Actual behavior
```
  6:1  error  The type 'foo' is undefined  jsdoc/no-undefined-types

✖ 1 problem (1 error, 0 warnings)
```

the typescript language server is able to resolve `foo` since i can go to definition on it, so this error is a false positive

## ESLint Config

<!--

What is the minimal config that reproduces the issue?

Please try to reproduce with only a single rule in your config (along
with the minimal number of options or settings that are needed to trigger
the error).

-->
```js
/** @type {import('eslint').Linter.Config} */
const config = {
  extends: [
    'plugin:@typescript-eslint/recommended',
  ],
  parser: '@typescript-eslint/parser',
  parserOptions: {
    ecmaVersion: 12,
    sourceType: 'module',
    project: './tsconfig.json',
  },
  rules: {
    'import/no-undefined-types': 'error',
  },
};

module.exports = config;
```

## ESLint sample

<!-- What code triggers the error? -->

```ts
declare global {
  const foo: number;
}

/**
 * {@link foo}
 */
export const bar = 1;
```

<!-- OR, if this is not triggered by a single rule, what is the
   problem and how can one reproduce? -->

## Environment

- Node version: 22.17.0
- ESLint version 8.57.1
- `eslint-plugin-jsdoc` version: 63.0.2

--- a/docs/rules/no-undefined-types.md
+++ b/docs/rules/no-undefined-types.md
@@ -1251,5 +1251,14 @@ class Test {
 }
 
 console.log(Test);
+
+declare global {
+  const foo: number;
+}
+
+/**
+ * {@link foo}
+ */
+export const bar = 1;
 ````
 

--- a/src/rules/noUndefinedTypes.js
+++ b/src/rules/noUndefinedTypes.js
@@ -340,6 +340,17 @@ export default iterateJsdoc(({
   /** @type {Set<string>} */
   const closedTypes = new Set();
 
+  const tsModuleVariables = scopeManager.scopes.find(({
+    type,
+  }) => {
+    // @ts-expect-error TS
+    return type === 'tsModule';
+  })?.variables.map(({
+    name,
+  }) => {
+    return name;
+  }) ?? [];
+
   const allDefinedTypes = new Set(globalScope.variables.map(({
     name,
   }) => {
@@ -431,6 +442,7 @@ export default iterateJsdoc(({
     .concat(typedefDeclarations)
     .concat(importTags)
     .concat(definedTypes)
+    .concat(tsModuleVariables)
     .concat(/** @type {string[]} */ (definedPreferredTypes))
     .concat((() => {
       // Other methods are not in scope, but we need them, and we grab them here

--- a/test/rules/assertions/noUndefinedTypes.js
+++ b/test/rules/assertions/noUndefinedTypes.js
@@ -2151,5 +2151,20 @@ export default /** @type {import('../index.js').TestCases} */ ({
         console.log(Test);
       `,
     },
+    {
+      code: `
+        declare global {
+          const foo: number;
+        }
+
+        /**
+         * {@link foo}
+         */
+        export const bar = 1;
+      `,
+      languageOptions: {
+        parser: typescriptEslintParser,
+      },
+    },
   ],
 });

